src/Decompiler/Loading/Loader.cs

```diff
@@ -313,43 +386,48 @@ public Program LoadRawImage(string filename, byte[] image, Address? addrLoad, Lo
         /// </summary>
         /// <param name="fileName"></param>
         /// <returns>a TypeLibrary instance, or null if the format of the file wasn't recognized.</returns>
-        public TypeLibrary? LoadMetadata(string fileName, IPlatform platform, TypeLibrary typeLib)
+        public TypeLibrary? LoadMetadata(RekoUri metadataUri, IPlatform platform, TypeLibrary typeLib)
         {
-            var rawBytes = LoadImageBytes(fileName, 0);
-            var mdLoader = FindImageLoader<MetadataLoader>(fileName, rawBytes);
+            var rawBytes = LoadImageBytes(metadataUri, 0);
+            var mdLoader = FindImageLoader<MetadataLoader>(metadataUri, rawBytes);
             if (mdLoader is null)
                 return null;
             var result = mdLoader.Load(platform, typeLib);
             return result;
         }
 
-        public ScriptFile? LoadScript(string fileName)
+        public ScriptFile? LoadScript(RekoUri scriptUri)
         {
-            var rawBytes = LoadImageBytes(fileName, 0);
-            return FindImageLoader<ScriptFile>(fileName, rawBytes);
+            var rawBytes = LoadImageBytes(scriptUri, 0);
+            return FindImageLoader<ScriptFile>(scriptUri, rawBytes);
         }
 
         /// <summary>
         /// Loads the contents of a file with the specified filename into an array 
         /// of bytes, optionally at the offset <paramref>offset</paramref>. No interpretation
         /// of those bytes is done.
         /// </summary>
-        /// <param name="fileName">File to open.</param>
+        /// <param name="imageUri">URI of the image whose bytes are to be loaded.</param>
         /// <param name="offset">The offset into the array into which the file will be loaded.</param>
         /// <returns>An array of bytes with the file contents at the specified offset.</returns>
-        public virtual byte[] LoadImageBytes(string fileName, int offset)
+        public virtual byte[] LoadImageBytes(RekoUri imageUri, int offset)
         {
+            // The initial fragment is always a file system location.
+            var fragments = UriTools.ParseUriIntoFragments(imageUri.ExtractString());
             var fsSvc = Services.RequireService<IFileSystemService>();
-            return fsSvc.ReadAllBytes(fileName);
+            var imageFile = imageUri.ToString();
+            return fsSvc.ReadAllBytes(fragments[0]);
```

Comment:
There is a challenge here. At the point where the archive is asked to load an image, it has all the metadata available to make a `Program` at that spot. Look for example on the MacOS CompactPro loader. All the information to load a resource fork is there, and so it makes sense to immediately load it. Otherwise, we have to register a magic number (there is none) or a file extension (there is none).

I like the approach that the archive loader gets a first chance of creating a `Program`. If it cannot (say, because the archive in question is a generic container, like an ISO or ZIP file) then it should return the new `Blob` image format back to the caller. The caller can then call `ILoader.LoadBinaryImage` on the bytes of `Blob`.